commit 436fc4aad3dcd616b665d95eb7cc409030086586
Author: Mikael Sundell <mikaelsundell@Mikaels-MacBook-Pro.local>
Date:   Mon Dec 29 19:14:02 2014 +0100

    added support for exr2 library names
    changed link libraries to use IlmBase_LIBRARY and OpenEXR_LIBRARY instead of manual library names
    Added IlmBase_LIBRARY to resolve missing Half symbols

diff --git a/cmake/modules/FindIlmBase.cmake b/cmake/modules/FindIlmBase.cmake
index a51cdd8..606ac53 100644
--- a/cmake/modules/FindIlmBase.cmake
+++ b/cmake/modules/FindIlmBase.cmake
@@ -52,7 +52,15 @@ endif()
 if("${IlmBase_VERSION}" VERSION_LESS "2.0.0")
   set(IlmBase_ALL_LIBRARIES Imath Half Iex IlmThread)
 else()
-  set(IlmBase_ALL_LIBRARIES Imath Half Iex IexMath IlmThread)
+  # handle new library names for 2.0.0
+  string(REPLACE "." "_" _IlmBase_VERSION ${IlmBase_VERSION})
+  string(SUBSTRING ${_IlmBase_VERSION} 0 3 _IlmBase_VERSION )
+  set(IlmBase_ALL_LIBRARIES
+    Half
+    Imath-${_IlmBase_VERSION} 
+    Iex-${_IlmBase_VERSION} 
+    IexMath-${_IlmBase_VERSION} 
+    IlmThread-${_IlmBase_VERSION})
 endif()
 foreach(ILMBASE_LIB ${IlmBase_ALL_LIBRARIES})
   string(TOUPPER ${ILMBASE_LIB} _upper_ilmbase_lib)
diff --git a/cmake/modules/FindOpenEXR.cmake b/cmake/modules/FindOpenEXR.cmake
index 40c1baa..2c8e83d 100644
--- a/cmake/modules/FindOpenEXR.cmake
+++ b/cmake/modules/FindOpenEXR.cmake
@@ -52,7 +52,18 @@ if(OpenEXR_INCLUDE_DIR AND EXISTS "${OpenEXR_INCLUDE_DIR}/OpenEXRConfig.h")
     endif()
 endif()
 
-find_library(OpenEXR_LIBRARY NAMES IlmImf libIlmImf HINTS ${_OpenEXR_HINT_LIB})
+if("${OpenEXR_VERSION}" VERSION_LESS "2.0.0")
+  set(IlmBase_ALL_LIBRARIES IlmImf libIlmImf)
+else()
+  # handle new library names for 2.0.0
+  string(REPLACE "." "_" _OpenEXR_VERSION ${OpenEXR_VERSION})
+  string(SUBSTRING ${_OpenEXR_VERSION} 0 3 _OpenEXR_VERSION )
+  set(IlmBase_ALL_LIBRARIES
+    IlmImf-${_OpenEXR_VERSION} 
+    libIlmImf-${_OpenEXR_VERSION})
+endif()
+
+find_library(OpenEXR_LIBRARY NAMES ${IlmBase_ALL_LIBRARIES} HINTS ${_OpenEXR_HINT_LIB})
 
 find_package(IlmBase QUIET)
 
diff --git a/lib/IlmCtl/CMakeLists.txt b/lib/IlmCtl/CMakeLists.txt
index 3672a43..1d177ab 100644
--- a/lib/IlmCtl/CMakeLists.txt
+++ b/lib/IlmCtl/CMakeLists.txt
@@ -21,8 +21,7 @@ add_library( IlmCtl ${DO_SHARED}
  CtlStdType.cpp
 )
 
-target_link_libraries( IlmCtl ${IlmBase_LDFLAGS_OTHER} )
-target_link_libraries( IlmCtl Iex IlmThread Half )
+target_link_libraries( IlmCtl ${IlmBase_LIBRARY} ${IlmBase_LDFLAGS_OTHER} )
 
 set_target_properties( IlmCtl PROPERTIES
   VERSION ${CTL_VERSION}
diff --git a/lib/IlmImfCtl/CMakeLists.txt b/lib/IlmImfCtl/CMakeLists.txt
index 47563df..f9c52c6 100644
--- a/lib/IlmImfCtl/CMakeLists.txt
+++ b/lib/IlmImfCtl/CMakeLists.txt
@@ -7,7 +7,7 @@ add_library( IlmImfCtl ${DO_SHARED}
   ImfCtlCopyFunctionArg.cpp
 )
 
-target_link_libraries( IlmImfCtl IlmCtl IlmImf Iex IlmThread Half )
+target_link_libraries( IlmImfCtl IlmCtl ${OpenEXR_LIBRARY} ${IlmBase_LIBRARY} )
 
 install( FILES ImfCtlApplyTransforms.h DESTINATION include/OpenEXR )
 
diff --git a/lib/dpx/CMakeLists.txt b/lib/dpx/CMakeLists.txt
index c8be26e..03def72 100644
--- a/lib/dpx/CMakeLists.txt
+++ b/lib/dpx/CMakeLists.txt
@@ -11,3 +11,4 @@ add_library( ctldpx
  dpx_rw.cc
 )
 
+target_link_libraries( ctldpx ${IlmBase_LIBRARY} )
